### ISSUE TITLE
api instructions

### ISSUE THREAD

**Issue by nkwsy** [theareba/mpesa-ruby#1]
Do you have any instructions for this? Does it work?

**Comment by theareba** [comment]
@nkwsy hi I'm open and was actually willing to create a ruby wrapper for Mpesa API. However due to complex nature of the docs, use of SOAP & availability of end-points at request of Safaricom made it impossible for me to accomplish that. What are your thoughts and suggestions on this? I would love to pull this off.

**Comment by theareba** [comment]
But if you were looking for an example of a gem that I've created to benchmark on, check this out https://github.com/theareba/thetvdb-ruby. Thanks.

**Comment by itskingori** [comment]
@nkwsy Looking at the code files, can't see any code. So I figure @theareba didn't get too far with this. Got a little exited at the possibility though. 😁

**Comment by theareba** [comment]
@itsmrwave this is something I would be open to working on in the near future. As stated, the problem with SOAP and nature of the docs made it difficult to kick start.

**Comment by itskingori** [comment]
Yeah, I concur.

It's just that an empty project creates expectation (shows up on search) ... which isn't met here. My personal opinion is this shouldn't be up, if it's empty. Or at a minimum, a prototype.

FYI, see check out [Project Mulla](http://kn9ts.github.io/project-mulla/) by @kn9ts. It's what you wanna do, but in JS (Node).

**Comment by kn9ts** [comment]
@theareba @itsmrwave Just a soft reminder that [Project Mulla](http://kn9ts.github.io/project-mulla/) is platform agnostic. it doesn't care about which language you use. You make HTTP REST requests to it from your backend servers programmed in your preferred language and it gets you back REST JSON payload responses.

**Comment by itskingori** [comment]
@kn9ts Yes, but it gives you two things to deploy (and maintain) ... this isn't a bad thing per se, just an observation.

I wanted to write a library when the API was launched as well but got turned off by the documentation and SOAP just like @theareba. Back then I wanted to build it as an extension of an app i.e. if included in the app, it makes the library's routes, controllers and all that available in the app context ... so it simply extends your existing app (deploy one thing instead of two).

That ☝️ ... is my preferred approach. But not to take away from anything you've done. 😁 ... it's still solid work!

**Comment by itskingori** [comment]
@theareba If you're interested ... I've started work on an `mpesa` gem ([see here](https://github.com/itsmrwave/mpesa-gem)). For use in any Ruby app/script.

After this I'll make `mpesa-rails` for a Rails app context. That will focus on `magical` stuff i.e. generating configurable routes in the main app, generating config files and ... more plugging into all that.

**Comment by theareba** [comment]
@itsmrwave this is awesome!!! Will checkout it out over the weekend. For `mpesa-rails` shouldn't that be part of the gem? So that you'd have generators in the gem that can create those routes, config files etc Or you mean creating an app that consumes your gem?

**Comment by theareba** [comment]
@itsmrwave been a while since I wrote a tutorial on my website http://www.victorareba.com/tutorials so if it turns out great I could write on this.

**Comment by itskingori** [comment]
@theareba I think it should be separate. Bundling Rails stuff in here assumes that only Rails people will use this. They should be able to use any framework i.e. Sinatra. Think of this as the core logic only. `mesa-rails` gem will have this as a dependency and inherit all it's functionality.
